libraries/pet_service/pet_service.lua

```diff
@@ -0,0 +1,176 @@
+            data.tp = p.pet_tp
+            if p.pet_index and p.pet_index ~= 0 then
+                data.active = true
+            else
+                data.active = false
+            end
+        end
+    end,
+    [{0x068}] = function(p)
+        if p.type == 4 then
+            data.owner_index = p.owner_index
+            data.owner_id = p.owner_id
+            data.index = p.pet_index
+            data.hp_percent = p.hp_percent
+            data.mp_percent = p.mp_percent
+            data.tp = p.pet_tp
+            data.target_id = p.target_id
+            data.name = p.pet_name
+            if p.pet_index and p.pet_index ~= 0 then
+                data.active = true
+            else
+                data.active = false
+            end
+        end
+    end,
+    [{0x044,0x12}] = function(p)
+
+        data.automaton.head.raw         = p.automaton_head
+        data.automaton.head.item_id     = p.automaton_head + 0x2000
+        data.automaton.frame.raw        = p.automaton_frame
+        data.automaton.frame.item_id    = p.automaton_frame + 0x2000
+        for i=0, 11 do
+            data.automaton.attachments[i].raw = p.attachments[i]
+            data.automaton.attachments[i].item_id = p.attachments[i] + 0x2100
+        end
+        ffi.copy(data.automaton._available_heads, p._available_heads, 4)
+        ffi.copy(data.automaton._available_frames, p._available_frames, 4)
+        ffi.copy(data.automaton._available_attach, p._available_attach, 32)
+        data.automaton.name             = p.pet_name
+        data.automaton.hp               = p.hp
+        data.automaton.hp_max           = p.hp_max
+        data.automaton.mp               = p.mp
+        data.automaton.mp_max           = p.mp_max
+        data.automaton.melee            = p.melee
+        data.automaton.melee_max        = p.melee_max
+        data.automaton.ranged           = p.ranged
+        data.automaton.ranged_max       = p.ranged_max
+        data.automaton.magic            = p.magic
+        data.automaton.magic_max        = p.magic_max
+        data.automaton.str              = p.str
+        data.automaton.str_modifier     = p.str_modifier
+        data.automaton.dex              = p.dex
+        data.automaton.dex_modifier     = p.dex_modifier
+        data.automaton.vit              = p.vit
+        data.automaton.vit_modifier     = p.vit_modifier
+        data.automaton.agi              = p.agi
+        data.automaton.agi_modifier     = p.agi_modifier
+        data.automaton.int              = p.int
+        data.automaton.int_modifier     = p.int_modifier
+        data.automaton.mnd              = p.mnd
+        data.automaton.mnd_modifier     = p.mnd_modifier
+        data.automaton.chr              = p.chr
+        data.automaton.chr_modifier     = p.chr_modifier
+
+        local active = data.active and (data.name == data.automaton.name)
+
+        data.automaton.active = active
+
+        if p.hp_max ~= 0 and active then
```

Comment:
And if we do the pointer based approach, this (and the following) check can be removed as well.

Reply:
I ended up just removing these in b0a3112. I originally did this because there is a slight minimum delay between each 0x044 (about 1sec?). Thinking on it more these detailed HP/MP auto stats seem better left as accurate, if slightly stale, rather than being estimated from low precision percents. This way addons can pick if speed or precision are more important.